### PR TITLE
BadUSB: qFlipper install script for MacOS

### PR DESCRIPTION
**Pull request by grugnoymeme**
# What's new

- BadUSB: qFlipper install script for MacOS

# Verification 

- Check `Install_qFlipper_macOS.txt`

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: assets/resources/badusb/Install_qFlipper_macOS.txt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+REM Author: 47LeCoste
+REM Target: macOS
+DELAY 300
+GUI SPACE
+DELAY 1000
+STRING Terminal
+DELAY 1000
+ENTER
+DELAY 1000
+STRING curl -o ~/Downloads/qFlipper.dmg -L https://update.flipperzero.one/builds/qFlipper/1.3.2/qFlipper-1.3.2.dmg
```
Comment (hedger) [inline assets/resources/badusb/Install_qFlipper_macOS.txt]: https://update.flipperzero.one/qFlipper/release/macos-amd64/dmg

**Review by hedger — CHANGES_REQUESTED** [review]
Please update URL

File: assets/resources/badusb/Install_qFlipper_macOS.txt (hunk 1 above)
Comment (grugnoymeme) [reply assets/resources/badusb/Install_qFlipper_macOS.txt]: yep!!

Comment (grugnoymeme) [reply assets/resources/badusb/Install_qFlipper_macOS.txt]: did it!

**Comment by hedger** [comment]
On my system, the delays didn't work even on the first try. Maybe rework it as a proper shell script? Or at least add ";" at commands' ends?
image

**Comment by grugnoymeme** [comment]
> On my system, the delays didn't work even on the first try. Maybe rework it as a proper shell script? Or at least add ";" at commands' ends?
> image

No i understand sorry. My fault i deleated an ENTER and A DELAY now i am correcting it

**Comment by grugnoymeme** [comment]
ok fixed, now it should work

**Comment by grugnoymeme** [comment]
yes. now i tested on two different macOS and it works for me !

**Review by hedger — DISMISSED** [review]

File: assets/resources/badusb/Install_qFlipper_macOS.txt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+REM Version 1.0 (Flipper Ducky)
+REM Target: macOS
+DELAY 300
+GUI SPACE
+DELAY 1000
+STRING Terminal
+DELAY 1000
+ENTER
+DELAY 1000
+STRING curl -L -o /tmp/qFlipper.dmg https://update.flipperzero.one/qFlipper/release/macos-amd64/dmg
```
Comment (skotopes) [inline assets/resources/badusb/Install_qFlipper_macOS.txt]: There are so many cases when it's not going to work.

Let's from scratch with couple thing in mind:
- Download time is unpredictable. 6 seconds are not going to help
- Delay based execution is cursed in a same way
- GUI SPACE - are not going to work in some cases (old keyboard layout shortcut)
- TAB not going to behave well

What must be done instead:
- Write standalone shell script that reliably works in all conditions
- on BadUSB side save this script into temporary file and execute it
- Minimize all unnecessary actions and delays

**Review by skotopes — CHANGES_REQUESTED** [review]

**Comment by skotopes** [comment]
Un-draft when ready.

File: assets/resources/badusb/Install_qFlipper_macOS.txt (hunk 2 above)
Comment (grugnoymeme) [reply assets/resources/badusb/Install_qFlipper_macOS.txt]: hey, i changed everything you asked. now it should be ok. let me know!!

**Comment by grugnoymeme** [comment]
sorry im pretty new  andi dont kown how to un-draft it!! :)

**Comment by grugnoymeme** [comment]
ok it sould be un-drafted.. right? sorry i'm pretty new .. XD

File: assets/resources/badusb/Install_qFlipper_macOS.txt (hunk 2 above)
Comment (grugnoymeme) [reply assets/resources/badusb/Install_qFlipper_macOS.txt]: i did not made it with a teporary shell script, but it would be ok this way as well, i think..

Comment (grugnoymeme) [reply assets/resources/badusb/Install_qFlipper_macOS.txt]: Im thinking if maybe running all with a "nohup" or a "screen -dm" before, could permit to quit the terminal and making the script continue working in background, showing nothing unless qFlipper starting after been moved in /Applications and ready to run .. let me know what do you think also about this :)

**Comment by grugnoymeme** [comment]
Anyway F4 should work for both: old and new systems.. in the old f4 is opening the apps launcher, and in the new one is opening spotlight.. in both cases you can type "terminal" and hit enter to open it with a mew window :)

**Comment by skotopes** [comment]
@grugnoymeme I'd do both Fn+F4 and F4

**Comment by grugnoymeme** [comment]
> @grugnoymeme I'd do both Fn+F4 and F4 

Is it possible to press Fn with badUSB from flipper? And how i should do?

Fn+F4
F4
(Rest of the script)??

**Comment by skotopes** [comment]
Yep, Fn+ not yet implemented. We'll do it a little bit later.

**Review by skotopes — APPROVED** [review]

**Comment by grugnoymeme** [comment]
> Yep, Fn+ not yet implemented. We'll do it a little bit later. 

Maybe we can also add a "nohup" in front of the main command to hide all the terminal process or it wouldn't be useful?

**Comment by skotopes** [comment]
No need, this script is not intended to run secretly, better to show user what it's doing.

**Pull request merged**